flaskbb/plugins/models.py

```diff
@@ -0,0 +1,81 @@
+# -*- coding: utf-8 -*-
+"""
+    flaskbb.plugins.models
+    ~~~~~~~~~~~~~~~~~~~~~~~
+
+    This module provides registration and a basic DB backed key-value
+    store for plugins.
+
+    :copyright: (c) 2017 by the FlaskBB Team.
+    :license: BSD, see LICENSE for more details.
+"""
+from sqlalchemy import UniqueConstraint
+from sqlalchemy.ext.associationproxy import association_proxy
+from sqlalchemy.orm.collections import attribute_mapped_collection
+
+from flaskbb.extensions import db
+from flaskbb.utils.database import CRUDMixin
+from flaskbb.utils.forms import generate_settings_form
+
+
+class PluginStore(CRUDMixin, db.Model):
+    id = db.Column(db.Integer, primary_key=True)
+    key = db.Column(db.Unicode(255), nullable=False)
+    value = db.Column(db.PickleType, nullable=False)
+    # Enum?
+    # Available types: string, integer, float, boolean, select, selectmultiple
+    value_type = db.Column(db.Unicode(20), nullable=False)
```

Comment:
Just noticed these extra nullable fields break the `association_proxy` on PluginRegistry. Should we keep that or remove it? At least it breaks for setting.

Reply:
mhh thats true.. what do you suggest? We could also revamp the way we create the setting forms but thats probably a different issue :).

Reply:
Mm, not sure. I like the idea of a simple KV store but if we need more information that what we can get out of that, it's not worth trying to make it work. I'm tempted to say we could infer the value's type, but that could lead to issues.